src/inversion_scripts/invert.py

```diff
@@ -207,15 +215,24 @@ def do_inversion(
     # Inverse of prior error covariance matrix, inv(S_a)
     Sa_diag = np.zeros(n_elements)
     Sa_diag.fill(prior_err**2)
-    
-    # if optimizing boundary conditions, adjust for it in the inversion
-    bc_idx = n_elements
-    if bc_optimization:
-        # add prior error for BCs
-        # as the last 4 elements of the diagonal
-        Sa_diag[-4:] = prior_err_bc**2
-        bc_idx -= 4
+
+    # If optimizing OH, adjust for it in the inversion
+    OH_idx = n_elements
+    if prior_err_oh > 0.0:
+        # add prior error for OH as the last element of the diagonal
+        Sa_diag[-1:] = (1/1000)*prior_err_oh**2 # try this
+        OH_idx -= 1
         
+    # If optimizing boundary conditions, adjust for it in the inversion
+    if bc_optimization:
+        # add prior error for BCs as the last 4 elements of the diagonal
+        if prior_err_oh > 0.0:
+            Sa_diag[-5:-2] = prior_err_bc**2
```

Comment:
I think that this indexing is not quite correct. If my understanding is correct, it should be selecting element (n-4),(n-3),(n-2),(n-1), whereas I think this indexing only selects (n-4),(n-3),(n-2).

See example:
```
>>> x = np.array([0,1,2,3,4,5,6,7,8,9])
>>> x[-5:-2]
array([5, 6, 7])
```

I think the fix is to make it `Sa_diag[-5:-1] = prior_err_bc**2`

Reply:
Now fixed in https://github.com/geoschem/integrated_methane_inversion/pull/157/commits/67f8f8a651ba6037f3cb882049755eb82ecb32c3